src/cargo/core/compiler/job_queue.rs

```diff
@@ -558,14 +581,21 @@ impl<'a, 'cfg> JobQueue<'a, 'cfg> {
 
     fn finish(
         &mut self,
+        id: u32,
         unit: &Unit<'a>,
         artifact: Artifact,
         cx: &mut Context<'_, '_>,
     ) -> CargoResult<()> {
         if unit.mode.is_run_custom_build() && cx.bcx.show_warnings(unit.pkg.package_id()) {
             self.emit_warnings(None, unit, cx)?;
         }
-        self.queue.finish(unit, &artifact);
+        let unlocked = self.queue.finish(unit, &artifact);
+        if let Some(t) = &mut self.timings {
```

Comment:
Oh so I actually liked the way this was earlier where you didn't have to write `if let` at all the callsites, could the `Option` be internal to the `timings` so from a calling perspective it's a bit more lightweight?